src/tools/structures/traitement-coselec.ts

```diff
@@ -0,0 +1,203 @@
+  const options = program.opts();
+  const pool = new Pool();
+  if (!ObjectId.isValid(options.structureId)) {
+    logger.error(`Veuillez renseigner un id structure.`);
+    return;
+  }
+  if (!options.quota || !options.numeroCoselec) {
+    logger.error(`Veuillez renseigner un quota et un numero COSELEC.`);
+    return;
+  }
+  const structure: IStructures = await app
+    .service(service.structures)
+    .Model.findOne({
+      _id: new ObjectId(options.structureId),
+    });
+  if (structure === null) {
+    logger.error(`Structure non trouvée.`);
+    exit();
+  }
+  const updatedAt = new Date();
+  const datePG = dayjs(updatedAt).format('YYYY-MM-DD');
+  const objectUpdated = {
+    $set: {
+      updatedAt,
+      coselecAt: updatedAt,
+      estLabelliseFranceServices: 'NON',
+    },
+    $push: {
+      coselec: {
+        nombreConseillersCoselec: Number(options.quota),
+        avisCoselec: 'POSITIF',
+        insertedAt: updatedAt,
+        numero: `COSELEC ${options.numeroCoselec}`,
+      },
+    },
+  };
+  const objectUpdatedMiseEnRelation = {
+    $set: {
+      'structureObj.updatedAt': updatedAt,
+      'structureObj.coselecAt': updatedAt,
+      'structureObj.estLabelliseFranceServices': 'NON',
+    },
+    $push: {
+      'structureObj.coselec': {
+        nombreConseillersCoselec: Number(options.quota),
+        avisCoselec: 'POSITIF',
+        insertedAt: updatedAt,
+        numero: `COSELEC ${options.numeroCoselec}`,
+      },
+    },
+  };
+  if (structure.statut === 'CREEE') {
+    Object.assign(objectUpdated.$set, {
+      'conventionnement.statut':
+        StatutConventionnement.CONVENTIONNEMENT_VALIDÉ_PHASE_2,
+      statut: 'VALIDATION_COSELEC',
+    });
+    Object.assign(objectUpdatedMiseEnRelation.$set, {
+      'structureObj.conventionnement.statut':
+        StatutConventionnement.CONVENTIONNEMENT_VALIDÉ_PHASE_2,
+      'structureObj.statut': 'VALIDATION_COSELEC',
+    });
+  }
+  if (
+    Number(options.quota) === 0 &&
+    structure.statut === 'VALIDATION_COSELEC'
+  ) {
+    const nbMiseEnRelationRecruter = await app
+      .service(service.misesEnRelation)
+      .Model.countDocuments({
+        statut: { $in: ['recrutee', 'finalisee', 'nouvelle_rupture'] },
+        'structure.$id': structure._id,
+      });
+    if (nbMiseEnRelationRecruter > 0) {
+      logger.error(
+        `La structure ${structure._id} possède des conseillers recrutés`,
+      );
+      exit();
+    }
+    Object.assign(objectUpdated.$set, {
+      statut: 'ABANDON',
+      userCreated: false,
+    });
+    Object.assign(objectUpdatedMiseEnRelation.$set, {
+      'structureObj.statut': 'ABANDON',
+      'structureObj.userCreated': false,
+    });
+    const userIds: IUser[] = await app
+      .service(service.users)
+      .Model.find({
+        'entity.$id': structure._id,
+        $and: [
+          { roles: { $elemMatch: { $eq: 'structure' } } },
+          { roles: { $elemMatch: { $eq: 'grandReseau' } } },
```

Comment:
pad besoin de $or dans le $and ?

Reply:
non sa remonte ceux qui ont le rôle structure et grandReseau, les deux roles obligatoirement